crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,208 @@
+
+        for package_path in &packages {
+            watched_dirs.push(package_path.clone());
+            println!("Watching {}", package_path.as_path().display());
+        }
+        let watched_dirs_clone = watched_dirs.clone();
+        let env_toml_path_clone = Arc::clone(&env_toml_path);
+        let env_toml_parent_clone = Arc::clone(&env_toml_parent);
+        let mut watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(event) = res {
+                    if matches!(
+                        event.kind,
+                        notify::EventKind::Create(_)
+                            | notify::EventKind::Modify(_)
+                            | notify::EventKind::Remove(_)
+                    ) {
+                        if let Some(path) = event.paths.first() {
+                            // Ignore temporary files
+                            if is_temporary_file(path) {
+                                return;
+                            }
+                            let env_toml_parent_abs = canonicalize_path(&env_toml_parent_clone);
+                            let env_toml_path_abs = canonicalize_path(&env_toml_path_clone);
+                            let parent_is_env_toml_parent =
+                                path.parent() == Some(env_toml_parent_abs.as_path());
+                            let path_is_env_toml = path == env_toml_path_abs.as_path();
+                            let parent_is_in_watched_dirs = is_parent_in_watched_dirs(
+                                &env_toml_parent_abs,
+                                &watched_dirs_clone,
+                            );
+                            // skip if the file is in the parent directory of environments.toml and it is not environments.toml
+                            if !(parent_is_env_toml_parent
+                                && !path_is_env_toml
+                                && !parent_is_in_watched_dirs)
+                            {
+                                println!("File changed: {path:?}");
+                                if let Err(e) = tx.blocking_send(Message::FileChanged) {
+                                    eprintln!("Error sending through channel: {e}");
+                                }
+                            }
+                        }
+                    }
+                }
+            })
+            .unwrap();
+        // Watch the parent directory of environments.toml
+        watcher.watch(env_toml_parent.as_path(), RecursiveMode::NonRecursive)?;
+        for package_path in watched_dirs {
+            watcher.watch(package_path.as_path(), RecursiveMode::Recursive)?;
+        }
+
+        if let Err(e) = self.build().await {
+            eprintln!("Build error: {e}");
+        }
+        println!("Watching for changes. Press Ctrl+C to stop.");
+
+        let rebuild_state_clone = Arc::clone(&rebuild_state);
```

Comment:
```suggestion
        let rebuild_state_clone = rebulid_state.clone();
```
Does this not work?

Reply:
Also do you need this at all? Can't you just `rebuild_state.clone()` below each time?

Reply:
Yes, I changed it, thanks.